lib/internal/http2/core.js

```diff
@@ -2387,13 +2381,14 @@ class Http2Server extends NETServer {
   }
 }
 
-function setupCompat(ServerRequest, ServerResponse, ev) {
+function setupCompat(ev) {
   if (ev === 'request') {
     debug('setting up compatibility handler');
     this.removeListener('newListener', setupCompat);
     this.on('stream', (stream, headers, flags, rawHeaders) => {
-      onServerStream.call(this, ServerRequest, ServerResponse,
-                          stream, headers, flags, rawHeaders);
+      onServerStream.bind(this)(this[kOptions].Http2ServerRequest,
+                                this[kOptions].Http2ServerResponse,
+                                stream, headers, flags, rawHeaders);
```

Comment:
You can just do `this.on('stream', onServerStream.bind(this[kOptions].Http2ServerRequest, this[kOptions].Http2ServerResponse));` The rest will work as expected.